charts/fission-all/templates/deployment.yaml

```diff
@@ -585,8 +587,18 @@ spec:
         env:
         - name: MESSAGE_QUEUE_TYPE
           value: nats-streaming
+        - name: MESSAGE_QUEUE_CLUSTER_ID
+          value: {{ .Values.nats.clusterID }}
+        - name: MESSAGE_QUEUE_QUEUE_GROUP
+          value: {{ .Values.nats.queueGroup }}
+        - name: MESSAGE_QUEUE_CLIENT_ID
+          value: {{ .Values.nats.clientID }}
         - name: MESSAGE_QUEUE_URL
+        {{- if not .Values.nats.url }}
           value: nats://{{ .Values.nats.authToken }}@nats-streaming:4222
+        {{- else }}
+          value: {{ .Values.nats.url }}
```

Comment:
```
value: nats://{{ .Values.nats.authToken }}@{{ .Values.nats.url }}
```

Reply:
From the example you attached, 
```
url: nats://nats-cluster.nats-io.svc.cluster.local:4222
```
How about we can add a an extra case for 
```
value: nats://{{ .Values.nats.authToken }}@{{ .Values.nats.url }}
```
if `authToken` is not empty. What do you think?

Reply:
One can include in the URL, but you may be right that a user might not think of that.  Hmm... the issue is that currently if you have an external id and don't have auth token (using user/password, for instance), or you have a different auth token, and include your token in the url. `nats.authToken` has a non-empty default. Also seems that "url" should mean entire url.

What about this: have two options: `nats.endpoint` and `nats.url` -- both blank by default. If `nats.enpoint` is set then the url is `nats://{{ .Values.nats.authToken }}@{{ .Values.nats.endpoint }}`; otherwise, if `nats.url` is set, the value is given by it, and if neither are set then as current?

Reply:
The endpoint is a bit useless because if the URL is not empty means user want to use external service. 

So it will be like
```
        {{- if not .Values.nats.url }}
          value: nats://{{ .Values.nats.authToken }}@nats-streaming:4222
        {{- else if .Values.nats.authToken }}
          value: nats://{{ .Values.nats.authToken }}@{{ .Values.nats.url }}
        {{- else }}
          value: {{ .Values.nats.url }}
        {{- end }}
```

Reply:
Hmm... so I was thinking:
```
{{- if and (not .Values.nats.url) (not .Values.nats.endpoint) }}
  value: nats://{{ .Values.nats.authToken }}@nats-streaming:4222
{{- else if not .Values.nats.url }}
  value: nats://{{ .Values.nats.authToken }}@{{ .Values.nats.endpoint }}
{{- else }}
   value: {{ .Values.nats.url }}
{{- end }}
```
The rationale being that `url` should always actually be a url; external service could be specified either with endpoint (for default service token) or url (not only auth token, but perhaps proxy ... so not necessarily even `nats://`).

But I'll patch according to what you say if I'm not convincing. Hmm... are you saying you want to get rid of `nats.external` and just use url (per your suggestion) or url or endpoint (per my suggestion) not only for nats url config, but also elsewhere for nats-streaming deployment and service, which are gated by `nats.external` now?

Reply:
oh, yes, I mean to get rid of `external`, sorry for that.

About `url` and `endpoint`, I personally think they are similar things and bring confusion to the user. Everyone has their own definition, and in the end they have to check how these two parameters work in the chart. 

How about we change it to `hostaddress` which is normally considered as the format of `<ip-or-dns>:<port>` to avoid the confusion?

The config will become

```
nats:
    hostaddress: nats-cluster.nats-io.svc.cluster.local:4222
    authToken: xxxxxxx
```

We can append the `nats://` in charts instead of putting it at values.yaml.

Reply:
This is good. Will send update this eve.

Hmm... oh -- but then we need to keep `nats.external` (for nats-streaming deployment/service gate), right?

Reply:
yes, if we change to `hostaddress` we will need external. Thanks for reminding me.

Reply:
btw, please add some comments to values.yaml as well so that people can quickly understand what `external` and other newly add parameter means.

Reply:
Very good